Consigment/Consigment/Consigment.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsigmentStruct
+{
+    public struct Consigment
+    {
+        int quantity;
+        public int Quantity
+        {
+            get => quantity;
+            set
+            {
+                if (value < 0)
+                    throw new ArgumentException("Количество единиц товара должно быть неотрицательным");
+                quantity = value;
+            }
+        }
+        double price;
+        public double Price
+        {
+            get => price;
+            set
+            {
+                if (value < 0)
+                    throw new ArgumentException("Количество единиц товара должно быть неотрицательным");
+                price = Math.Round(value, 2);
+            }
+        }
+        public double Cost
+        {
+            get => Math.Round((Price * Quantity),2);
+        }
+        public Consigment(int quantity, double price) : this()
+        {
+            Quantity = quantity;
+            Price = price;
+        }
+        public override string ToString() => $"{Quantity} шт. по {Price} руб.\"";
+        public override bool Equals(object obj)
+        {
+            if (obj is Consigment)
+                return (Quantity == ((Consigment)obj).Quantity) && (Price == ((Consigment)obj).Price);
+            throw new ArgumentException("Объект для сравнения не является партией товара");
+        }
+        public override int GetHashCode() => Cost.GetHashCode();
```

Comment:
Функция реализована неверно. Стоимость партии не определяет однозначно партию. Поэтому хеш-код стоимости не может использоваться для хеш-кода партии товара. Лучше было реализовать этот метод как в примере к заданию. Например,
```suggestion
       public override int GetHashCode()
        {
            int hash = 17;
            const int prime = 31;
            hash = hash * prime + quantity.GetHashCode();
            hash = hash * prime + price.GetHashCode();

            return hash;
        }
```